src/platforms/posix/drivers/df_isl_wrapper/df_isl_wrapper.cpp

```diff
@@ -0,0 +1,389 @@
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * @file df_isl_wrapper.cpp
+ * Driver to access the ISL of the DriverFramework.
+ *
+ * @author Nicolas de Palezieux <ndepal@gmail.com>
+ */
+
+#include <px4_config.h>
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <stdint.h>
+#include <stddef.h>
+#include <stdlib.h>
+#include <string.h>
+#include <math.h>
+#include <unistd.h>
+#include <px4_getopt.h>
+#include <errno.h>
+#include <string>
+
+#include <systemlib/perf_counter.h>
+#include <systemlib/err.h>
+
+#include <drivers/drv_range_finder.h>
+
+#include <uORB/uORB.h>
+#include <uORB/topics/subsystem_info.h>
+#include <uORB/topics/distance_sensor.h>
+
+#include <board_config.h>
+
+#include <isl29501/isl29501.hpp>
+#include <DevMgr.hpp>
+
+
+extern "C" { __EXPORT int df_isl_wrapper_main(int argc, char *argv[]); }
+
+using namespace DriverFramework;
+
+
+class DfISLWrapper : public ISL
+{
+public:
+	DfISLWrapper();
+	~DfISLWrapper();
+
+
+	/**
+	 * Start automatic measurement.
+	 *
+	 * @return 0 on success
+	 */
+	int		start();
+
+	/**
+	 * Stop automatic measurement.
+	 *
+	 * @return 0 on success
+	 */
+	int		stop();
+
+private:
+	int _publish(struct range_sensor_data &data);
+
+	orb_advert_t		_range_topic;
+
+	int			_orb_class_instance;
+
+	// perf_counter_t		_range_sample_perf;
+
+};
+
+DfISLWrapper::DfISLWrapper(/*enum Rotation rotation*/) :
+	ISL(ISL_DEVICE_PATH),
+	_range_topic(nullptr),
+	_orb_class_instance(-1)
+{
+}
+
+DfISLWrapper::~DfISLWrapper()
+{
+}
+
+int DfISLWrapper::start()
+{
+
+	struct distance_sensor_s d;
+	_range_topic = orb_advertise_multi(ORB_ID(distance_sensor), &d,
+					   &_orb_class_instance, ORB_PRIO_DEFAULT);
+
+	int ret;
+	ret = ISL::init();
+
+	if (ret != 0) {
+		PX4_ERR("ISL init fail: %d", ret);
+		return ret;
+	}
+
+	ret = ISL::start();
+
+	if (ret != 0) {
+		PX4_ERR("ISL start fail: %d", ret);
+		return ret;
+	}
+
+	return 0;
+}
+
+int DfISLWrapper::stop()
+{
+	/* Stop sensor. */
+	int ret = ISL::stop();
+
+	if (ret != 0) {
+		PX4_ERR("ISL stop fail: %d", ret);
+		return ret;
+	}
+
+	return 0;
+}
+
+int DfISLWrapper::_publish(struct range_sensor_data &data)
+{
+	if (!_range_topic) {
+		return 1;
+	}
+
+	struct distance_sensor_s d;
+
+	memset(&d, 0, sizeof(d));
+
+	d.timestamp = hrt_absolute_time();
+
+	d.min_distance = float(ISL_MIN_DISTANCE); /* m */
+
+	d.max_distance = float(ISL_MAX_DISTANCE); /* m */
+
+	d.current_distance = float(data.dist);
```

Comment:
Do you mean this correction: https://github.com/3drobotics/DriverFramework/commit/2adc6b50b8bf63855f71e62da334601547e7b78c#diff-4aab583654b51f98c08d57b7a171a770R268